src/TrustedHostsNetworkResolver.php

```diff
@@ -59,9 +59,22 @@
  *   ['x-forwarded-for', 'forwarded', ...],
  * );
  * ```
+ *
+ * @psalm-type HostData = array{ip?:string, host?: string, by?: string, port?: string|int, protocol?: string, httpHost?: string}
+ * @psalm-type ProtocolHeadersData = array<string, array<non-empty-string, array<array-key, string>>|callable>
+ * @psalm-type TrustedHostData = array{
+ *     'hosts': array<string>,
+ *     'ipHeaders': array<string>,
+ *     'urlHeaders': array<string>,
+ *     'portHeaders': array<string>,
+ *     'trustedHeaders': array<string>,
+ *     'protocolHeaders': ProtocolHeadersData,
+ *     'hostHeaders': array<string>
```

Comment:
Instead of `array<string>` need use `list<string>` or `array<array-key,string>`. [Docs](https://psalm.dev/docs/annotating_code/type_syntax/array_types/)